### PR TITLE
dependabot + updated buf build + updated CLA build + regen protos

### PR DESCRIPTION
**Pull request by vroldanbet**
Bunch of stuff here:
- updated actions used
- added dependabot
- removed duplicate actions on different events
- addressed failures with buf gen & diff after moving to more recent version of grpc
- now using authzed/actions for buf generation - also updated the version used
- generated proto code with latest API, so it includes `lookupsubjects`
- updated CLA job with latest centralized GH action

### CODE REVIEW

**Comment by vroldanbet** [comment]
This is ready to go! I've commented the `lookupsubjects` test because it will fail, spicedb 1.12 hasn't been released yet.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -77,30 +40,35 @@ jobs:
...
           python-version: "3.10"
       - name: "Setup Python Environment"
         run: "pip install -U pip virtualenv"
       - name: "Install Homebrew & gRPC"
         run: |
           bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+          echo "/home/linuxbrew/.linuxbrew/bin" >> $GITHUB_PATH
           brew install grpc
-      - uses: "bufbuild/buf-setup-action@v0.6.0"
+          ls /home/linuxbrew/.linuxbrew/bin
```
Comment (samkim) [inline .github/workflows/test.yaml:57]: Nice to have for visibility but we can remove these later if the buf gen works consistently for the next few updates

**Review by samkim — APPROVED** [review]
LGTM

File: .github/workflows/test.yaml (hunk 1 above)
Comment (vroldanbet) [reply .github/workflows/test.yaml:57]: agreed! I also left some extra flags in the `buf.gen.yaml` that hopefully will help troubleshoot things if we get in trouble again 😅

**Pull request merged**